data/impl/RecordSet.js

```diff
@@ -164,28 +164,22 @@ export class RecordSet {
     }
 
     createRecord(raw, records, parent) {
-        const {store} = this,
-            {idSpec} = store;
+        const {store} = this;
 
-        if (store.processRawData) store.processRawData(raw);
-
-        raw.id = isString(idSpec) ? raw[idSpec] : idSpec(raw);
-
-        throwIf(
-            isNil(raw.id),
-            "Record has a null/undefined ID. Use the 'LocalStore.idSpec' config to resolve a unique ID for each record."
-        );
+        let data = raw;
+        if (store.processRawData) {
+            data = store.processRawData(raw);
+            throwIf(!data, 'processRawData should return an object. If writing/editing, be sure to return a clone!');
```

Comment:
Should we enforce processRawData always returning a new object? Is there a likely/important use case where we have processRawData making decisions to *not* edit some objects (and it would be terribly inefficient to force it to clone anyway)?